pkg/collector/vg_collector.go

```diff
@@ -0,0 +1,145 @@
+*/
+
+package collector
+
+import (
+	"github.com/openebs/lvm-localpv/pkg/lvm"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/klog"
+)
+
+/*
+vgCollector collects all the necessary metrics related to volume group
+*/
+type vgCollector struct {
+	vgSizeMetric              *prometheus.Desc
+	vgFreeMetric              *prometheus.Desc
+	vgLvCountMetric           *prometheus.Desc
+	vgPvCountMetric           *prometheus.Desc
+	vgMaxLvMetric             *prometheus.Desc
+	vgMaxPvMetric             *prometheus.Desc
+	vgSnapCountMetric         *prometheus.Desc
+	vgMissingPvCountMetric    *prometheus.Desc
+	vgMetadataCountMetric     *prometheus.Desc
+	vgMetadataUsedCountMetric *prometheus.Desc
+	vgMetadataFreeMetric      *prometheus.Desc
+	vgMetadataSizeMetric      *prometheus.Desc
+	vgPermissionsMetric       *prometheus.Desc
+	vgAllocationPolicyMetric  *prometheus.Desc
+}
+
+func NewVgCollector() prometheus.Collector {
+	return &vgCollector{
+		vgFreeMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "free_size_bytes"),
+			"LVM VG free size in bytes",
+			[]string{"name"}, nil,
+		),
+		vgSizeMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "total_size_bytes"),
+			"LVM VG total size in bytes",
+			[]string{"name"}, nil,
+		),
+		vgLvCountMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "lv_count"),
+			"Number of LVs in VG",
+			[]string{"name"}, nil,
+		),
+		vgPvCountMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "pv_count"),
+			"Number of PVs in VG",
+			[]string{"name"}, nil,
+		),
+		vgMaxLvMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "max_lv_count"),
+			"LMaximum number of LVs allowed in VG or 0 if unlimited",
+			[]string{"name"}, nil,
+		),
+		vgMaxPvMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "max_pv_count"),
+			"Maximum number of PVs allowed in VG or 0 if unlimited",
+			[]string{"name"}, nil,
+		),
+		vgSnapCountMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "snap_count"),
+			"Number of snapshots in VG",
+			[]string{"name"}, nil,
+		),
+		vgMissingPvCountMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "missing_pv_count"),
+			"Number of PVs in VG which are missing",
+			[]string{"name"}, nil,
+		),
+		vgMetadataCountMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "mda_count"),
+			"Number of metadata areas on this VG",
+			[]string{"name"}, nil,
+		),
+		vgMetadataUsedCountMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "mda_used_count"),
+			"Number of metadata areas in use on this VG",
+			[]string{"name"}, nil,
+		),
+		vgMetadataFreeMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "mda_free_size_bytes"),
+			"Free metadata area space for this VG in bytes",
+			[]string{"name"}, nil,
+		),
+		vgMetadataSizeMetric: prometheus.NewDesc(prometheus.BuildFQName("lvm", "vg", "mda_total_size_bytes"),
+			"Size of smallest metadata area for this VG in bytes",
```

Comment:
what does this mean?

Reply:
This is the metadata size that the VG contains.